api/apps/api/src/modules/scenario-specification/application/assign-candidate-specification.handler.ts

```diff
@@ -0,0 +1,32 @@
+import {
+  CommandHandler,
+  EventPublisher,
+  IInferredCommandHandler,
+} from '@nestjs/cqrs';
+import { AssignCandidateSpecification } from '../command/assign-candidate-specification.command';
+import { SpecificationId } from '../specification.id';
+import { ScenarioSpecificationRepo } from './scenario-specification.repo';
+
+@CommandHandler(AssignCandidateSpecification)
+export class AssignCandidateSpecificationHandler
+  implements IInferredCommandHandler<AssignCandidateSpecification> {
+  constructor(
+    private readonly scenarioSpecificationRepo: ScenarioSpecificationRepo,
+    private readonly eventPublisher: EventPublisher,
+  ) {}
+
+  async execute({
+    specificationId,
+    scenarioId,
+  }: AssignCandidateSpecification): Promise<void> {
+    const scenarioSpecification = this.eventPublisher.mergeObjectContext(
+      await this.scenarioSpecificationRepo.findOrCreate(scenarioId),
+    );
+    scenarioSpecification.assignCandidateSpecification(
+      new SpecificationId(specificationId),
+    );
+    await this.scenarioSpecificationRepo.save(scenarioSpecification);
```

Comment:
shouldn't it be in a transaction?

Reply:
No need so I believe. Empty scenario specification don't have to be persisted (during `findOrCreate`)

Reply:
it's find or create, so may lead to lost updates